### DISCUSSION TITLE
Unable to configure gateway using kernel parameters on metal platform

### DISCUSSION THREAD

**Discussion by daghemo** [siderolabs/talos#9982] [Q&A] [answered]
Unable to configure gateway using kernel parameters on metal platform. As reported by @omniproc on #8959. Tried on 1.8.3 and 1.9.0.
Added network configuration as shown on https://www.talos.dev/v1.8/talos-guides/install/virtualized-platforms/proxmox/, that is:

`ip=10.105.155.21::10.105.155.30:255.255.255.240:pve1:eth0:off`

Both the IP and the hostname shows up in the dash, after boot. The gateway not. Connectivity is reported as failed.

Any idea?
IMHO #8959 should be re-opened.

**Comment by smira** [comment]
A bit unrelated, but on Proxmox I would recommend to use https://www.talos.dev/v1.9/talos-guides/install/cloud-platforms/nocloud/ to configure networking, it's way easier.

**Comment by smira** [comment, marked as answer]
I tested booting up with your settings and there are no issues, and gateway is indeed added:

```
[   14.739109] [talos] created route {"component": "controller-runtime", "controller": "network.RouteSpecController", "destination": "default", "gateway": "10.105.155.30", "table": "main", "link": "eth0", "priority": 1024, "family": "inet4"}
```

Make sure you do `net.ifnames=0` if you want to use `eth0` (vs. predictable interface names by default).

Either way, on Proxmox you have serial console (boot with `console=ttyS0`) to debug further.

**Comment by daghemo** [comment]
I confirm that the following works:

`net.ifnames=0 ip=10.105.155.21::10.105.155.30:255.255.255.240:pve1:eth0:off`

I saw “eth0” in the “Network Config” window and had not thought that predictable names were used by default.
Thanks!
